images/miq-app/docker-assets/container-deploy-common.sh

```diff
@@ -102,10 +100,33 @@ if [[ -z $APP_VERSION || -z $SCHEMA_VERSION || -z $IMAGE_VERSION ]]; then
   echo "${PV_DEPLOY_INFO_FILE} is incomplete, one or more required variables are undefined"
   exit 1
 else
-  echo "PV_APP_VERSION=${APP_VERSION}" > "${PV_DEPLOY_INFO_FILE}"
-  echo "PV_SCHEMA_VERSION=${SCHEMA_VERSION}" >> "${PV_DEPLOY_INFO_FILE}"
-  echo "PV_IMG_VERSION=${IMAGE_VERSION}" >> "${PV_DEPLOY_INFO_FILE}"
-  echo "PV_DEPLOYMENT_DATE=${DEPLOYMENT_DATE}" >> "${PV_DEPLOY_INFO_FILE}"
+  case "${DEPLOYMENT_STATUS}" in
+    redeployment)
+    ;;
+    upgrade)
+    # PV_DEPLOY_INFO_FILE must exist on upgrades
+    [ ! -f "${PV_DEPLOY_INFO_FILE}" ] && echo "ERROR: Something seems wrong, ${PV_DEPLOY_INFO_FILE} could not be found" && exit 1
+    # Backup existing PV_DEPLOY_INFO_FILE
+    cp "${PV_DEPLOY_INFO_FILE}" "${PV_BACKUP_DIR}/backup_${PV_BACKUP_TIMESTAMP}"
+    cp "${PV_DEPLOY_INFO_FILE}" "${PV_DEPLOY_INFO_FILE}~"
+    # Re-write file with upgraded deployment info
+    echo "PV_APP_VERSION=${APP_VERSION}" > "${PV_DEPLOY_INFO_FILE}"
+    echo "PV_SCHEMA_VERSION=${SCHEMA_VERSION}" >> "${PV_DEPLOY_INFO_FILE}"
+    echo "PV_IMG_VERSION=${IMAGE_VERSION}" >> "${PV_DEPLOY_INFO_FILE}"
+    echo "PV_DEPLOYMENT_DATE=${DEPLOYMENT_DATE}" >> "${PV_DEPLOY_INFO_FILE}"
```

Comment:
${PV_DEPLOY_INFO_FILE}~ is secondary backup copy of the original file, this file will remain since it is stored on the PV, I also have the primary backup copy of ${PV_DEPLOY_INFO_FILE} original file in the backup directory.